velox/exec/fuzzer/WindowFuzzer.cpp

```diff
@@ -68,20 +68,69 @@ void WindowFuzzer::addWindowFunctionSignatures(
   }
 }
 
-std::tuple<std::string, bool> WindowFuzzer::generateFrameClause() {
-  auto frameType = [](int value) -> const std::string {
+void WindowFuzzer::addKRangeFrameColumnToInput(
+    std::vector<RowVectorPtr>& input,
+    core::WindowNode::BoundType& frameBoundType,
+    const std::string& columnName) {
+  // For frames with k RANGE PRECEDING/FOLLOWING, Velox requires the application
+  // to add columns with the range frame boundary value computed according
+  // to the frame type.
+  // If the frame is k PRECEDING :
+  // frame_boundary_value = current_order_by - k (for ascending ORDER BY)
+  // frame_boundary_value = current_order_by + k (for descending ORDER BY)
+  // If the frame is k FOLLOWING :
+  // frame_boundary_value = current_order_by + k (for ascending ORDER BY)
+  // frame_boundary_value = current_order_by - k (for descending ORDER BY)
+  auto computeOffset = [&](const core::WindowNode::BoundType& frameBoundType,
+                           const int64_t& offsetValue,
+                           const bool& ascending) -> int64_t {
+    if (frameBoundType == core::WindowNode::BoundType::kCurrentRow) {
+      return 0;
+    }
+    if (frameBoundType == core::WindowNode::BoundType::kPreceding) {
+      return ascending ? -offsetValue : offsetValue;
+    } else {
+      return ascending ? offsetValue : -offsetValue;
+    }
+  };
+
+  auto size = vectorFuzzer_.getOptions().vectorSize;
+  velox::test::VectorMaker vectorMaker{pool_.get()};
+
+  for (auto j = 0; j < FLAGS_num_batches; ++j) {
+    int64_t offsetValue =
+        boost::random::uniform_int_distribution<int64_t>(1, 10)(rng_);
+    // Setting ascending boolean to true since we're using ascending row_number
+    // column as the order by column
+    int64_t offset = computeOffset(frameBoundType, offsetValue, true);
+    auto column = vectorMaker.flatVector<int64_t>(
+        size, [offset](auto row) { return row + offset; });
+
+    auto rowVectorBatch = input[j];
```

Comment:
This can be done just once for the entire input I think. Why are you doing this per batch ?

Reply:
@aditi-pandit From my understanding, the input variable contains `FLAGS_num_batches` number of batches and in each batch contains the columns. That is why I have to re-create the batch with the new k0 column added to it and then re-assign it back to input[j] for all j in `FLAGS_num_batches`. Please correct me if I understood this incorrectly.